src/util/yp_constant_pool.c

```diff
@@ -13,7 +13,7 @@ yp_constant_id_list_init(yp_constant_id_list_t *list) {
 bool
 yp_constant_id_list_append(yp_constant_id_list_t *list, yp_constant_id_t id) {
     if (list->size >= list->capacity) {
-        list->capacity = list->capacity == 0 ? 1 : list->capacity * 2;
+        list->capacity = list->capacity == 0 ? 8 : list->capacity * 2;
         list->ids = realloc(list->ids, sizeof(yp_constant_id_list_t) * list->capacity);
```

Comment:
```suggestion
        list->ids = realloc(list->ids, sizeof(yp_constant_id_t) * list->capacity);
```
It would cause allocating more than necessary.
(and it's a bit sad a C compiler doesn't warn about that)

Reply:
Ugh, good catch. The compiler catches it if you cast, but not otherwise.